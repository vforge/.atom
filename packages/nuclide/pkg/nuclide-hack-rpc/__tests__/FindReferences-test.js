'use strict';

var _FindReferences;

function _load_FindReferences() {
  return _FindReferences = require('../lib/FindReferences');
}

var _nuclideTestHelpers;

function _load_nuclideTestHelpers() {
  return _nuclideTestHelpers = require('../../nuclide-test-helpers');
}

/**
 * Copyright (c) 2015-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the LICENSE file in
 * the root directory of this source tree.
 *
 *  strict-local
 * @format
 */

const projectRoot = '/test/';
const file1Path = '/test/file1.php';
const file2Path = '/test/file2.php';

describe('FindReferences', () => {
  it('convertReferences', () => {
    expect((0, (_FindReferences || _load_FindReferences()).convertReferences)([{
      name: '\\TestClass::testFunction',
      filename: file1Path,
      line: 13,
      char_start: 5,
      char_end: 7
    }, {
      name: '\\TestClass::testFunction',
      filename: file2Path,
      line: 11,
      char_start: 1,
      char_end: 3
    }], projectRoot)).toMatchSnapshot();
  });
});