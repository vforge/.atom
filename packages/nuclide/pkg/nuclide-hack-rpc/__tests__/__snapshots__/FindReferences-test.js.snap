// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindReferences convertReferences 1`] = `
Object {
  "baseUri": "/test/",
  "referencedSymbolName": "TestClass::testFunction",
  "references": Array [
    Object {
      "name": null,
      "range": Range {
        "end": Point {
          "column": 7,
          "row": 12,
        },
        "start": Point {
          "column": 4,
          "row": 12,
        },
      },
      "uri": "/test/file1.php",
    },
    Object {
      "name": null,
      "range": Range {
        "end": Point {
          "column": 3,
          "row": 10,
        },
        "start": Point {
          "column": 0,
          "row": 10,
        },
      },
      "uri": "/test/file2.php",
    },
  ],
  "type": "data",
}
`;
